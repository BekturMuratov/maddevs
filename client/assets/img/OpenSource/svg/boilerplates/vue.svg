<svg xmlns="http://www.w3.org/2000/svg" width="336" height="125" fill="none" xmlns:v="https://vecta.io/nano"><g clip-path="url(#A)"><path d="M206.784.001l-38.797 67.203L129.191.001H-.008l167.996 290.997L335.984.001h-129.2z" fill="#41b883"/><path d="M206.785.001l-38.797 67.203L129.191.001h-62l100.798 174.595L268.786.001h-62.001z" fill="#34495e"/></g><defs><clipPath id="A"><path fill="#fff" d="M0 0h336v291H0z"/></clipPath></defs></svg>