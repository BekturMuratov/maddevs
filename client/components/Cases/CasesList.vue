<template>
  <div class="container">
    <section class="cases-list">
      <!-- Card -->
        <CasesListItem
          :video-file-name="sjmc.video"
          :logo="sjmc.logo"
          :subtitle="sjmc.subtitle"
          :title="sjmc.title"
          :desc="sjmc.desc"
        />
      <!-- Card -->
      <NuxtLink
        to="/"
        class="cases-list_card cases-list_card-big"
      >
        <!-- Video BG -->
        <video
          loop="true"
          muted="true"
          autoplay="true"
        >
          <source
            :src="$getMediaFromS3('/videos/main.ef19480.mp4')"
            type="video/mp4"
          >
          Your browser does not support the video tag.
        </video>
        <!-- END Video BG -->
        <!-- Card info -->
        <div class="cases-list_card-info">
          <img
            :src="$getMediaFromS3(`/images/Cases/${logo.folder}/svg/${logo.file}.svg`)"
            :alt="logo.alt"
          >
          <span>Foodtech</span>
          <h3>Contactless Delivery Service</h3>
          <p>Mad Devs created the Namba Food delivery service from scratch. The solution orchestrates feature-rich apps for couriers, end-users, and business owners.</p>
          <NuxtLink
            to="/"
          >
            Explore
          </NuxtLink>
        </div>
        <!-- END Card info  -->
      </NuxtLink>
      <!-- Card -->
      <NuxtLink
        to="/"
        class="cases-list_card cases-list_card-small"
      >
        <!-- Video BG -->
        <video
          loop="true"
          muted="true"
          autoplay="true"
        >
          <source
            :src="$getMediaFromS3('/videos/yourcast-banner.mp4')"
            type="video/mp4"
          >
          Your browser does not support the video tag.
        </video>
        <!-- END Video BG -->
        <!-- Card info -->
        <div class="cases-list_card-info">
          <img
            :src="$getMediaFromS3(`/images/Cases/${logo.folder}/svg/${logo.file}.svg`)"
            :alt="logo.alt"
          >
          <span>Content Streaming</span>
          <h3>The Evolution of Yourcast.TV</h3>
          <p>Mad Devs developed a secure and private video streaming service and live video broadcasting cinema to provide entertainment experience to isolated groups feeling homesick.</p>
          <NuxtLink
            to="/"
          >
            Explore
          </NuxtLink>
        </div>
        <!-- END Card info  -->
      </NuxtLink>
      <!-- Card -->
      <NuxtLink
        to="/"
        class="cases-list_card cases-list_card-small"
      >
        <!-- Video BG -->
        <video
          loop="true"
          muted="true"
          autoplay="true"
        >
          <source
            :src="$getMediaFromS3('/videos/main.ef19480.mp4')"
            type="video/mp4"
          >
          Your browser does not support the video tag.
        </video>
        <!-- END Video BG -->
        <!-- Card info -->
        <div class="cases-list_card-info">
          <img
            :src="$getMediaFromS3(`/images/Cases/${logo.folder}/svg/${logo.file}.svg`)"
            :alt="logo.alt"
          >
          <span>BYOD</span>
          <h3>Veeqo – platform for e-commerce</h3>
          <p>Mad Devs created the Namba Food delivery service from scratch. The solution orchestrates feature-rich apps for couriers, end-users, and business owners.</p>
          <NuxtLink
            to="/"
          >
            Explore
          </NuxtLink>
        </div>
        <!-- END Card info  -->
      </NuxtLink>
      <!-- Card -->
      <NuxtLink
        to="/"
        class="cases-list_card cases-list_card-big"
      >
        <!-- Video BG -->
        <video
          loop="true"
          muted="true"
          autoplay="true"
        >
          <source
            :src="$getMediaFromS3('/videos/godee-case-main-video.mp4')"
            type="video/mp4"
          >
          Your browser does not support the video tag.
        </video>
        <!-- END Video BG -->
        <!-- Card info -->
        <div class="cases-list_card-info">
          <img
            :src="$getMediaFromS3(`/images/Cases/${logo.folder}/svg/${logo.file}.svg`)"
            :alt="logo.alt"
          >
          <span>Transportation</span>
          <h3>Convenient shuttle bus service</h3>
          <p>Mad Devs helped GoDee with developing feature-rich software to re-invent public mobility by building new smart ways of a daily commute.</p>
          <NuxtLink
            to="/"
          >
            Explore
          </NuxtLink>
        </div>
        <!-- END Card info  -->
      </NuxtLink>
    </section>
    <!-- NOTE: Temporarily commented -->
    <!-- <button
      class="cases-list_see-more"
    >
      <span>See More</span> ↓
    </button> -->
  </div>
</template>

<script>
import CasesListItem from '@/components/Cases/CasesListItem'

export default {
  name: 'CasesList',
  components: { CasesListItem },
  data() {
    return {
      logo: {
        width: 259,
        height: 82,
        folder: 'nambafood',
        file: 'nambafood-logo',
        alt: '',
      },

      nambafood: '/videos/main.ef19480.mp4',
      yourcast: '/videos/yourcast-banner.mp4',
      veequ: '/videos/main.ef19480.mp4',
      godee: '/videos/godee-case-main-video.mp4',

      sjmc: {
        video: '/videos/sjmc/sjmc-main-video.b35a387.mp4',
        subtitle: 'SJMC',
        title: 'Sir John Monash Centre',
        desc: 'Mad Devs engineers helped Sir John Monash Centre to enhance and maintain the BYOD multimedia technology.',
        logo: {
          width: 259,
          height: 82,
          folder: 'nambafood',
          file: 'nambafood-logo',
          alt: '',
        },
      },
    }
  },

  methods: {
    play() {
      this.$refs.video.play()
    },

    pause() {
      this.$refs.video.pause()
    },
  },
}
</script>

<style lang="scss" scoped>
  .cases-list {
    width: 100%;
    display: grid;
    grid-template-columns: repeat(4, 1fr);
    grid-auto-flow: dense;
    gap: 30px;

    @media screen and (max-width: 375px) {
      gap: 8px;
    }

    &_card {
      width: 100%;
      max-width: 100%;
      height: 605px;
      max-height: 605px;
      position: relative;
      border-radius: 2px;
      padding: 40px;
      box-sizing: border-box;
      overflow: hidden;
      text-decoration: none;

      a {
        text-decoration: none;
      }

      &-full {
        grid-column: auto / span 4;
      }

      &-big {
        grid-column: auto / span 3;
      }

      &-middle {
        grid-column: auto / span 2;
      }

      &-small {
        grid-column: auto / span 1;
      }

      @media screen and (max-width: 375px) {
        padding: 24px;
        grid-column: auto / span 4 !important;
      }

      video {
        position: absolute;
        top: 0;
        left: 0;
        z-index: -1;
        height: 100%;
        width: 100%;
        object-fit: cover;
      }

      &::before {
        content: '';
        width: 100%;
        height: 100%;
        position: absolute;
        top: 0;
        left: 0;
        z-index: 0;
        background-color: rgba(0, 0, 0, 0.25);
        user-select: none;
        pointer-events: none;
      }

      &-info {
        width: 100%;
        height: 100%;
        display: flex;
        flex-direction: column;
        align-items: flex-start;
        justify-content: flex-start;
        position: relative;
        z-index: 1;
        color: #F4F4F4;

        * {
          color: inherit;
          font-family: 'Inter', sans-serif;
        }

        img {
          max-width: 105px;
          height: auto;
          margin-bottom: auto;
        }

        > span {
          text-transform: uppercase;
          letter-spacing: 0.28em;
          font-size: 10px;
          font-weight: normal;
          line-height: 140%;
          margin-bottom: 16px;
        }

        h3 {
          max-width: 350px;
          font-size: 40px;
          letter-spacing: -0.04em;
          font-weight: bold;
          line-height: 100%;
          margin-bottom: 16px;

          @media screen and (max-width: 375px) {
            font-size: 32px;
          }
        }

        p {
          max-width: 500px;
          font-size: 16px;
          letter-spacing: -0.013em;
          line-height: 140%;
          font-weight: normal;
          margin-bottom: 16px;
        }

        > a {
          font-size: 16px;
          line-height: 100%;
          text-decoration: none;
          padding: 8px;
          background-color: #fff;
          color: #000;
          border-radius: 4px;

          @media screen and (max-width: 375px) {
            font-size: 14px;
          }
        }
      }
    }

    &_see-more {
      color: #EC1C24;
      font-weight: normal;
      font-size: 16px;
      line-height: 100%;
      letter-spacing: -0.013em;
      margin: 50px auto 0;
      font-family: 'Inter', sans-serif;
      background-color: transparent;
      border: 0;
      cursor: pointer;

      span {
        position: relative;

        &::after {
          content: '';
          width: 100%;
          height: 1px;
          display: block;
          position: absolute;
          bottom: 0;
          left: 0;
          background-color: #EC1C24;
        }
      }
    }
  }

  .container {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    justify-self: start;
    padding-bottom: 120px;

    @media screen and (max-width: 375px) {
      padding: 0;
    }
  }
</style>
