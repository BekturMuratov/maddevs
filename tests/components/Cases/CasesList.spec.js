import CasesList from '@/components/Cases/CasesList.vue'
import { render } from '@testing-library/vue'
import '../../__mocks__/intersectionObserverMock'

const stubs = ['CasesListItem', 'NuxtLink']

const props = {
  limit: 3,
}

describe('CasesList component', () => {
  it('should render correctly', () => {
    const { container } = render(CasesList, {
      stubs,
    })

    expect(container).toMatchSnapshot()
  })

  it('should render three cards', () => {
    const { getAllByTestId } = render(CasesList, {
      stubs,
      props,
    })

    expect(getAllByTestId('test-case-card')).toHaveLength(props.limit)
  })
})
