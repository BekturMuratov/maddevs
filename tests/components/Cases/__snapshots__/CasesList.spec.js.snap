// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CasesList component should render correctly 1`] = `
<div>
  <div
    class="container"
  >
    <section
      class="cases-list"
    >
      <caseslistitem-stub
        class="cases-list_sjmc"
        data-testid="test-case-card"
        desc="Mad Devs engineers helped Sir John Monash Centre to enhance and maintain the BYOD multimedia technology."
        link="/case-studies/sir-john-monash-centre/"
        logo="[object Object]"
        poster="/images/Cases/sjmc/jpg/sjmc-poster.jpg"
        subtitle="BYOD"
        title="Sir John Monash Centre"
        titletag="h3"
        videofilename="/videos/sjmc/sjmc-main-video.b35a387.mp4#t=2"
        width="full"
      />
      <caseslistitem-stub
        class="cases-list_nambafood"
        data-testid="test-case-card"
        desc="Mad Devs created the Namba Food delivery service from scratch. The solution orchestrates feature-rich apps for couriers, end-users, and business owners."
        link="/case-studies/namba-food/"
        logo="[object Object]"
        poster="/images/Cases/nambafood/jpg/nambafood-poster.jpg"
        subtitle="Foodtech"
        title="Contactless Delivery Service"
        titletag="h3"
        videofilename="/videos/main.ef19480.mp4#t=1"
        width="big"
      />
      <caseslistitem-stub
        class="cases-list_yourcast"
        data-testid="test-case-card"
        desc="Mad Devs developed a secure and private video streaming service and live video broadcasting cinema to provide entertainment experience to isolated groups feeling homesick."
        link="/case-studies/yourcast/"
        logo="[object Object]"
        poster="/images/Cases/yourcast/jpg/yourcast-poster.jpg"
        subtitle="Content Streaming"
        title="The Evolution of Yourcast.TV"
        titletag="h3"
        videofilename="/videos/yourcast-banner.mp4#t=1"
        width="small"
      />
      <caseslistitem-stub
        class="cases-list_godee"
        data-testid="test-case-card"
        desc="Mad Devs helped GoDee with developing feature-rich software to re-invent public mobility by building new smart ways of a daily commute."
        link="/case-studies/godee/"
        logo="[object Object]"
        poster="/images/Cases/godee/jpg/godee-poster.jpg"
        subtitle="Transportation"
        title="Convenient shuttle bus service"
        titletag="h3"
        videofilename="/videos/godee-case-main-video.mp4#t=1"
        width="full"
      />
      <caseslistitem-stub
        class="cases-list_itc"
        data-testid="test-case-card"
        desc="A web app customized for the delivery of export management coaching missions that contributes to the development of intra-regional and international trade in Central Asia"
        link="/case-studies/R4TCA-web-application/"
        logo="[object Object]"
        poster="/images/Cases/itc/jpg/itc-banner.jpg"
        subtitle="Digital transformation"
        title="Ready4Trade Central Asia"
        titletag="h3"
        videofilename="/videos/itc-banner.mp4#t=1"
        width="big"
      />
      <caseslistitem-stub
        class="cases-list_veeqo"
        data-testid="test-case-card"
        desc="Our DevOps team optimized the infrastructure for Veeqo, an inventory and shipping e-commerce platform, and made it highly efficient, smooth, and cost-effective for our partners."
        link="/case-studies/veeqo/"
        logo="[object Object]"
        poster="/images/Cases/veeqo/svg/veeqo-banner-bg.svg"
        subtitle="Logistics"
        title="Optimization for Veeqo"
        titletag="h3"
        videofilename="/videos/veeqo-banner.mp4#t=1"
        width="small"
      />
    </section>
     
    <nuxtlink-stub
      class="cases-list_see-more"
      to="/case-studies/"
    >
      <span>
        See More
      </span>
    </nuxtlink-stub>
  </div>
</div>
`;
